app/scripts/services/authorization.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, $log, $interval, DataService, APIService){
+    function AuthorizationService() {
+      this.rules = null;
+      this.intervalTimer = null;
+      this.forceReload = false;
+      this.canCreateSomeResource = false;
+      this.wasCanCreateSomeResourceSet = false;
+      this.projectNameRules = null;
+    }
+
+    AuthorizationService.prototype.reviewUserRules = function($scope) {
```

Comment:
I know we do this in other places, but we shouldn't pass scope to services. Services shouldn't know anything about a controller's scope.